### DISCUSSION TITLE
Bingham Model

### DISCUSSION THREAD

**Discussion by Arminsj** [Anura3D/Anura3D_OpenSource#10] [Q&A] [answered]
Hello,
I have a question about the Bingham model Code.
In the last part of the code (line 119 UMATBingham),  the liquid pressure is updated using volumetric strain for the liquid pressure lower than the cavitation threshold. However, according to the scientific manual, the stress should be updated using volumetric strain for the pressures above the cavitation threshold (page 60, v2021). I wonder if there is a problem or not. 
"
          COMPUTEPRESSURE = (LIQUIDPRESSURE.LT.LIQUIDPRESSURECAVITATION)

          
          IF(COMPUTEPRESSURE) THEN
             
            LIQUIDPRESSUREINCREMENT = BULKMODULUSLIQUID  * DSTRANVOL

            LIQUIDPRESSURE = LIQUIDPRESSURE + LIQUIDPRESSUREINCREMENT

          
            IF(LIQUIDPRESSURE .GT. LIQUIDPRESSURECAVITATION) THEN
              LIQUIDPRESSURE=LIQUIDPRESSURECAVITATION
              STRESS(1:3) = LIQUIDPRESSURE
              STRESS(4:NTENS) = 0.0
              RETURN
            END IF  
     "

**Comment by murphy-john** [comment, marked as answer]
Hello @Arminsj , Thanks for your question. I asked this question at our developers meeting and hopefully one of our group will get back to you with an answer. 

We removed the constitutive laws from the scientific manuals because they needed to be updated. It's possible this was simply an typo in the manual. It may be as simple as the sign convention for tension and compression for stresses (& pressures) was different between the manual and how it was implemented. The Anura3D source code, stresses (∴ pressures) use the following convention: tension (positive) and compression (negative).
